### PR TITLE
Delete DumpIR code

### LINKED ISSUE #818: Remove COMPlus_JitDumpIR
As described [here](https://github.com/dotnet/runtime/issues/760#issuecomment-564745955), consider removing unused and unmaintained `COMPlus_JitDumpIR` code.

category:implementation
theme:jit-coding-style
skill-level:beginner
cost:small

### PR DESCRIPTION
**Pull request by mikedn**
I keep running into this code in my `GT_LIST` removal work so... delete.

I kept `cLoopIR` but changed it do use normal tree dump. Never used it but it seems potentially useful and it's not a lot of code.

I also kept `cTreeFlags`. It's a ton of code and I'm not sure if it's up to date but it seems useful as dumps do not include all flags and trying to figure out which flags is set by looking at the hex number in debugger is a bit of pain.

Fixes #818

### CODE REVIEW

**Review by sandreenko — APPROVED** [review]
LGTM, thanks for doing this.
cc @dotnet/jit-contrib

**Review by CarolEidt — APPROVED** [review]
LGTM also. Given that this is just debug-only dumping code, I see no reason to wait for the CI issues to be resolved before merging this.

**Pull request merged**
